pulp_puppet_plugins/pulp_puppet/plugins/importers/importer.py

```diff
@@ -80,9 +81,21 @@ def upload_unit(self, repo, type_id, unit_key, metadata, file_path, conduit, con
         try:
             report = upload.handle_uploaded_unit(repo, type_id, unit_key, metadata, file_path,
                                                  conduit)
+
```

Comment:
I can do that. Question about how to proceed which direction I should proceed in:

1. Have the parent exception `ExtractionException` inherit from `PulpCodedException`
2. Have these two exceptions inherit from `PulpCodedException` and drop `ExtractionException`
3. Drop all these exceptions and just raise a coded error message like we do [here](https://git.io/vyLoo).

Here's a link to where we define these exceptions:

https://github.com/pulp/pulp_puppet/blob/master/pulp_puppet_plugins/pulp_puppet/plugins/importers/metadata.py